mangaki/mangaki/algo/fma.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import pywFM
+import os
+from scipy.sparse import coo_matrix
+
+from mangaki.algo.recommendation_algorithm import (RecommendationAlgorithm,
+                                                   register_algorithm)
+
+
+@register_algorithm('fma', {'rank': 20})
+class MangakiFMA(RecommendationAlgorithm):
+    def __init__(self, rank=20, nb_iterations=10):
+        super().__init__()
+        self.rank = rank
+        self.nb_iterations = nb_iterations
+        self.fm = None
+
+    def load(self, filename):
+        backup = super().load(filename)
+        self.mu = backup.mu
+        self.W = backup.W
+        self.V = backup.V
+        self.V2 = backup.V2
+
+    @property
+    def is_serializable(self):
+        return False  # Not yet
+
+    def prepare_fm(self, X):
+        nb_samples = len(X)
+        user_ids = X[:, 0]
+        work_ids = X[:, 1]
+        # For the k-th user_id-work_id pair, we need (k, user_id) and (k, N + work_id), so two copies of range(nb_samples)
+        rows = list(range(nb_samples)) * 2
```

Comment:
Wonder if we could do it faster using NumPy built-ins for `range` and so on?

Reply:
Good question. I will provide a better implementation, actually, anyway.